homeassistant/components/doorbird/button.py

```diff
@@ -20,18 +20,25 @@
 class DoorbirdButtonEntityDescription(ButtonEntityDescription):
     """Class to describe a Doorbird Button entity."""
 
-    press_action: Callable[[DoorBird, str], Coroutine[Any, Any, bool]]
+    press_action: Callable[[ConfiguredDoorBird, str], Coroutine[Any, Any, bool | None]]
 
 
 RELAY_ENTITY_DESCRIPTION = DoorbirdButtonEntityDescription(
     key="relay",
-    translation_key="relay",
-    press_action=lambda device, relay: device.energize_relay(relay),
+    press_action=lambda door_station, relay: door_station.device.energize_relay(relay),
 )
-IR_ENTITY_DESCRIPTION = DoorbirdButtonEntityDescription(
-    key="ir",
-    translation_key="ir",
-    press_action=lambda device, _: device.turn_light_on(),
+BUTTON_DESCRIPTIONS: tuple[DoorbirdButtonEntityDescription, ...] = (
+    DoorbirdButtonEntityDescription(
+        key="__ir_light__",
```

Comment:
Meh.. shouldn't we just migrate instead?

Reply:
We should definitely do a migration but I want to add tests to everything before doing that.

I have a lot of legacy to cleanup here and want to also handle Martin's comment in https://github.com/home-assistant/core/pull/121114#discussion_r1668171539 before I do that